### ISSUE TITLE
[OSX] Place nvidia-cuda install behind flag (was: nvidia-cuda not available for Mojave)

### ISSUE THREAD

**Issue by randyzwitch** [heavyai/heavydb#326]
I just tried to use the `scripts/mapd-deps-osx.sh` script, which fails at the nvidia-cuda step:

```bash
Error: Cask nvidia-cuda depends on macOS release being one of [10.12, 10.13], but you are running release 10.14.
```

If this is only required for GPU support, we should make a flag or remove the installation step for these drivers, as it doesn't seem like NVIDIA/Apple are working on providing this

**Comment by andrewseidl** [comment]
Yeah, it's only needed to build with GPU support. Should be safe to default it off these days :(

**Comment by psilvest** [comment]
This issue has been inactive for a while, and the project has changed substantially. If this still occurs on the latest release, please add reproduction steps and environment details.
